hibernate-reactive-core/src/test/java/org/hibernate/reactive/schema/NumericValidationTest.java

```diff
@@ -0,0 +1,185 @@
+/* Hibernate, Relational Persistence for Idiomatic Java
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * Copyright: Red Hat Inc. and Hibernate Authors
+ */
+package org.hibernate.reactive.schema;
+
+import static org.hibernate.tool.schema.JdbcMetadaAccessStrategy.GROUPED;
+import static org.hibernate.tool.schema.JdbcMetadaAccessStrategy.INDIVIDUALLY;
+
+import java.math.BigDecimal;
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.Id;
+
+import org.hibernate.cfg.Configuration;
+import org.hibernate.reactive.BaseReactiveTest;
+import org.hibernate.reactive.containers.DatabaseConfiguration;
+import org.hibernate.reactive.provider.Settings;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import io.vertx.ext.unit.TestContext;
+
+public abstract class NumericValidationTest extends BaseReactiveTest {
+
+	public static class IndividuallyNumericValidationTestBase extends NumericValidationTest {
+
+		@Override
+		protected Configuration constructConfiguration(String hbm2DdlOption) {
+			final Configuration configuration = super.constructConfiguration( hbm2DdlOption );
+			configuration.setProperty( Settings.HBM2DDL_JDBC_METADATA_EXTRACTOR_STRATEGY, INDIVIDUALLY.toString() );
+			return configuration;
+		}
+	}
+
+	public static class GroupedNumericValidationTestBase extends NumericValidationTest {
+
+		@Override
+		protected Configuration constructConfiguration(String hbm2DdlOption) {
+			final Configuration configuration = super.constructConfiguration( hbm2DdlOption );
+			configuration.setProperty( Settings.HBM2DDL_JDBC_METADATA_EXTRACTOR_STRATEGY, GROUPED.toString() );
+			return configuration;
+		}
+	}
+
+	protected Configuration constructConfiguration(String hbm2DdlOption) {
+		Configuration configuration = constructConfiguration();
+		configuration.setProperty( Settings.HBM2DDL_AUTO, hbm2DdlOption );
+		configuration.setProperty( Settings.DEFAULT_CATALOG, "hreact" );
```

Comment:
Is setting the catalog part of the corner case? Because I don't think it's going to work for all databases

Reply:
good point. Will tweak that test.

Reply:
I know that there was an issue where not setting the catalog wil lcause a `NullPointerException`. Iif you rebase on top of [this pr](https://github.com/hibernate/hibernate-reactive/pull/970), that issue should be solved.

Reply:
It is the dialect's implementation of `#getNameQualifierSupport` that determines what *may* be required.

If `NameQualifierSupport#CATALOG` is returned, then `Settings.DEFAULT_CATALOG` may need to be defined.

If `NameQualifierSupport#SCHEMA` is returned, then `Settings.DEFAULT_SCHEMA` may need to be defined.

If `NameQualifierSupport#BOTH` is returned, then both  `Settings.DEFAULT_CATALOG` and `Settings.DEFAULT_SCHEMA` may need to be defined.

I think it depends on the particular database whether it is needed and whether `GROUPED` strategy is required. I'm not sure what happens if, for example `#getNameQualifierSupport` returns `NameQualifierSupport#CATALOG` and `Settings.DEFAULT_SCHEMA` is provided. ORM may simply ignore the setting.

@dreab8, may be able to shed more light on this.

Reply:
If I remember correctly if the db does not support schema and we provide a schema in the mapping of a table we have errors not sure about the Settings, but I may be wrong...We need to test this behaviour.